crates/typst/src/foundations/scope.rs

```diff
@@ -109,13 +114,27 @@ fn cannot_mutate_constant(var: &str) -> HintedString {
 fn unknown_variable(var: &str) -> HintedString {
     let mut res = HintedString::new(eco_format!("unknown variable: {}", var));
 
+    if var.contains('-') {
+        res.hint(eco_format!(
+            "if you meant to use subtraction, try adding spaces around the minus sign(s): `{}`",
+            var.replace('-', " - ")
+        ));
+    }
+
+    res
+}
+
+#[cold]
+fn unknown_variable_math(var: &str, in_global: bool) -> HintedString {
+    let mut res = HintedString::new(eco_format!("unknown variable: {}", var));
+
     if matches!(var, "none" | "auto" | "false" | "true") {
         res.hint(eco_format!(
-            "if you meant to use a literal, try adding a hash before it"
+            "if you meant to use a literal, try adding a hash before it: `#{var}`",
         ));
-    } else if var.contains('-') {
+    } else if in_global {
         res.hint(eco_format!(
-            "if you meant to use subtraction, try adding spaces around the minus sign",
+            "`{var}` is not available in math, add a hash before it: `#{var}`",
```

Comment:
```suggestion
            "`{var}` is not available in directly math, try adding a hash before it: `#{var}`",
```
- I think "directly" makes it a bit clearer that it _is_ available, just with hash.
- The "add" -> "try adding" is maybe unnecessary, but I feel like with all hints that make suggestions, it might be good to be cautious with the wording

Reply:
I'm not a native English speaker so not sure about this
but I think it should be either "not directly available" or "not available in math directly" or "not available directly in math"

Reply:
yeah typo. I wanted to say "not available directly in math"